cabot/cabotapp/models.py

```diff
@@ -349,16 +350,20 @@ def is_silenced(self, now=None):
 
 class Service(CheckGroupMixin):
 
+    @transaction.atomic()
     def update_status(self):
+        vals = Service.objects.select_for_update().filter(pk=self.pk).values_list('deleted', flat=True)
```

Comment:
nit: a comment explaining this would be nice

Reply:
I was in a rush when I wrote this, but I think this fixes it for MySQL because it locks the service until the transaction completes.

So if
* process A calls `update_service()`
* process B calls `service.deleted = True; service.save(); service.delete()` between `vals` being evaluated and the end of the function

Then process B's `service.save()` will block until process A returns.

If it didn't block, then:
* Process B would generate the list of snapshots to delete
* Right after, process A adds another snapshot
* Process B finishes cleaning up its snapshot list
* Process B goes to delete the `service` row, but fails due to a dangling foreign key from `snapshot -> service` because process B never saw the new snapshot

I'll add this as a comment too...